core/src/main/java/org/mskcc/cbio/portal/util/StructuralVariantUtil.java

```diff
@@ -0,0 +1,162 @@
+        structuralVariant.setNormalPairedEndReadCount(TabDelimitedFileUtil.getPartInt(getColumnIndex(StructuralVariantUtil.NORMAL_PAIRED_END_READ_COUNT), parts));
+        structuralVariant.setTumorPairedEndReadCount(TabDelimitedFileUtil.getPartInt(getColumnIndex(StructuralVariantUtil.TUMOR_PAIRED_END_READ_COUNT), parts));
+        structuralVariant.setNormalSplitReadCount(TabDelimitedFileUtil.getPartInt(getColumnIndex(StructuralVariantUtil.NORMAL_SPLIT_READ_COUNT), parts));
+        structuralVariant.setTumorSplitReadCount(TabDelimitedFileUtil.getPartInt(getColumnIndex(StructuralVariantUtil.TUMOR_SPLIT_READ_COUNT), parts));
+        structuralVariant.setAnnotation(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.ANNOTATION), parts));
+        structuralVariant.setBreakpointType(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.BREAKPOINT_TYPE), parts));
+        structuralVariant.setCenter(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.CENTER), parts));
+        structuralVariant.setConnectionType(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.CONNECTION_TYPE), parts));
+        structuralVariant.setEventInfo(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.EVENT_INFO), parts));
+        structuralVariant.setVariantClass(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.VARIANT_CLASS), parts));
+        structuralVariant.setLength(TabDelimitedFileUtil.getPartInt(getColumnIndex(StructuralVariantUtil.LENGTH), parts));
+        structuralVariant.setComments(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.COMMENTS), parts));
+        structuralVariant.setExternalAnnotation(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.EXTERNAL_ANNOTATION), parts));
+        structuralVariant.setDriverFilter(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.DRIVER_FILTER), parts));
+        structuralVariant.setDriverFilterAnn(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.DRIVER_FILTER_ANNOTATION), parts));
+        structuralVariant.setDriverTiersFilter(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.DRIVER_TIERS_FILTER), parts));
+        structuralVariant.setDriverTiersFilterAnn(TabDelimitedFileUtil.getPartString(getColumnIndex(StructuralVariantUtil.DRIVER_TIERS_FILTER_ANNOTATION), parts));
+        return structuralVariant;
+    }
+
+    public int getColumnIndex(String colName) {
+        Integer index = this.columnIndexMap.get(colName.toLowerCase());
+        if (index == null) {
+            index = -1;
+        }
+        return index;
+    }
+
+    public Boolean hasRequiredStructuralVariantFields(StructuralVariant record) {
+        return !record.getSite1EnsemblTranscriptId().equalsIgnoreCase(TabDelimitedFileUtil.NA_STRING) &&
```

Comment:
This is fine if we want to filter out all records which are not complete with the full set of information for visualization. I think @n1zea144 suggested that we could have a two-level approach though ... where we let through records which have none of these fields filled in; and also let through records which have all of these fields filled in. We would only filter records which have *at least one, but not all* of these fields filled in. This is because the simpler records which were missing transcripts and exons still were being presented in the frontend --- just without the extra visualization showing the exons and the junction between members of the fusion. So these primitive records would just show that there was *some* kind of fusion which happened between to the two genes, but with no details. I think @n1zea144 wanted to allow these primitive fusions to go through ... and also let the fully detailed fusions go through. We would filter the "a little extra detail but not enough" records. (In my mind we could drop the partial extra details and turn it into a primitive record, but I think that is the kind of data fixing we are now trying to avoid)

Reply:
@sheridancbio Oh I see, basically allow records with all or "nothing" filled in for those key columns. I just want to confirm with @n1zea144 as well since the (now removed) file format documentation for SV data says that the site 1 and 2 exons and ensembl transcript id fields [are required](https://github.com/cBioPortal/cbioportal/blob/da5740b5c450758f0a90e3aa5cd6274c18882c0a/docs/File-Formats.md#structural-variants-data).